src/date-utils.es6

```diff
@@ -150,6 +150,22 @@ const DateUtils = {
     }
     return moment(date)
   },
+
+  getTimeFormat(opts) {
```

Comment:
If you've got parameter destructuring available to you, I reckon you could clean this code up a bit like so:
```js
getTimeFormat({ seconds, timeZone, upperCase }) {
  const use24HourClock = NylasEnv.config.get('core.workspace.use24HourClock')
  let timeFormat = use24HourClock ? "HH:mm" : "h:mm"
  if (seconds) {
    timeFormat += ':ss'
  }
  if (!use24HourClock && upperCase) {
    timeFormat += ' A'
  }
  if (timeZone) {
    timeFormat += ' z'
  }

  return timeFormat
}
```
While there's still a number of conditionals that could be pulled out and made in to a set of functions that are applied to the original string, but hey why overcomplicate.